halo2_proofs/src/plonk/prover.rs

```diff
@@ -147,6 +147,7 @@ where
         k: u32,
         current_phase: sealed::Phase,
         advice: Vec<Polynomial<Assigned<F>, LagrangeCoeff>>,
+        unblinded_advice: HashSet<usize>,
```

Comment:
from what I can tell  this shouldn't affect the determinism of key generation -- as its only used to check for set membership here. And accesses on `HashSet` will be faster than on a `BTreeSet` (mostly relevant for large circuits with lots of cols)

Reply:
Since it's not put into `PinnedConstraintSystem`, so ut wouldn't make anything non-deterministic I think.

As to put it into `PinnedConstraintSystem` or not, I think it's not necessary, I feel this is more like a prover strategy but with current prover api there is no way to achieve without breaking change, so we do this in constraint system instead.